### PR TITLE
Fix keyboard events

### PR DESCRIPTION
**Pull request by Strikeskids**
Add wrapper modules that extract useful keyboard information from key events. Performs `keyCode` to `code` normalization.

### CODE REVIEW

**Comment by hhugo** [comment]
Thank you

File: lib/dom_html.ml [hunk 1]
```diff
@@ -166,13 +166,19 @@ end
...
   method altKey : bool t readonly_prop
   method shiftKey : bool t readonly_prop
   method ctrlKey : bool t readonly_prop
   method metaKey : bool t readonly_prop
+  method location : int readonly_prop
+
+  method key : js_string t optdef readonly_prop
+  method code : js_string t optdef readonly_prop
+
+  method which : int optdef readonly_prop
```
Comment (vasilisp) [inline lib/dom_html.ml]: This breaks Eliom. We need [multiple inheritance on `event`, `mouseEvent`, and `keyboardEvent`](https://github.com/ocsigen/eliom/blob/d24b99e13bb957de4e83a31636dffec72490dd83/src/lib/eliom_runtime.shared.mli#L70-L74). The new `which` method renders them incompatible.

Comment (hhugo) [inline lib/dom_html.ml]: It this multiple inheritance actually needed ?

Comment (hhugo) [inline lib/dom_html.ml]: I'm already looking at the fix for ocaml 4.01

Comment (vasilisp) [inline lib/dom_html.ml]: The multiple inheritance is needed to have a uniform internal representation of event handlers. The resulting "biggest event" is used in contravariant position in handlers, so we can use any of the events in its place.

I will see if we can separate them by providing multiple constructors in place of [`CE_client_closure`](https://github.com/ocsigen/eliom/blob/d24b99e13bb957de4e83a31636dffec72490dd83/src/lib/eliom_runtime.shared.mli#L60).

Comment (vasilisp) [inline lib/dom_html.ml]: Fixed by ocsigen/eliom@4edaf22 .

**Pull request merged**